adoc/chapters/programming_interface.adoc

```diff
@@ -5540,358 +5423,318 @@ queue, the sub-buffer's alignment must be compatible with both the primary
 queue's device and the secondary queue's device, otherwise this exception is
 thrown.
 
-Must throw an [code]#exception# with the [code]#errc::invalid# error code if
-[code]#b# is a sub-buffer.
+_Throws:_
 
-|====
+* An [code]#exception# with the [code]#errc::invalid# error code if [code]#b#
+is a sub-buffer.
 
+* An [code]#exception# with the [code]#errc::invalid# error code if
+[code]#subRange# corresponds to a a non-contiguous region of [code]#b#.
 
+- An [code]#exception# with the [code]#errc::invalid# error code if the sum
+of [code]#baseIndex# and [code]#subRange# in any dimension exceeds the parent
+buffer [code]#b# size [code]#bufferRange# in that dimension.
+'''
```

Comment:
This contradicts the _Preconditions_.  If something is a _Precondition_ then violating it results in undefined behavior.  Each of these conditions should either be a precondition or be a condition in which an exception is thrown.